dlt/common/schema/typing.py

```diff
@@ -165,6 +167,7 @@ class TTableSchema(TypedDict, total=False):
     columns: TTableSchemaColumns
     resource: Optional[str]
     table_format: Optional[TTableFormat]
+    table_index_type: Optional[TTableIndexType]
```

Comment:
@rudolfix we can switch between 2 index types (https://learn.microsoft.com/en-us/azure/synapse-analytics/sql-data-warehouse/sql-data-warehouse-tables-index) and we have a new table hint here. I am wondering if we can re-use table_format or wether this needs to be a separate hint. I am not quite sure.

Reply:
this looks like a case for destination adapter. it sets per-destination options on a resource using x- hints. please look into:
https://github.com/dlt-hub/dlt/blob/devel/dlt/destinations/adapters.py
and bigquery adapter that @Pipboyguy is doing: https://github.com/dlt-hub/dlt/issues/855

Reply:
I rewrote the code to make use of a destination adapter. Looked at `qdrant_adapter` and `weaviate_adapter` for inspiration. The `synapse_adapter` is slightly different because it works with a table hint, not column hints like `qdrant` and `weaviate`. Since `x-table-index-type` is not defined on `TResourceHints`, I have to ignore mypy's `typeddict-unknown-key` errors. Is this what you had in mind @rudolfix?